test/vault/client/http_test.clj

```diff
@@ -44,3 +46,66 @@
 
       (is (= @connection-attempt api-endpoint)
           (str "should attempt to auth with: " api-endpoint)))))
+
+(deftest authenticate-via-k8s
+  (let [tmp-token-file (fn [token-contents]
+                         (let [f (doto (File/createTempFile "k8s-" ".token")
```

Comment:
Thanks for adding tests! These are actually a good example of how avoiding the file coupling would simplify different usage patterns.